CitationStyleLanguagePlugin.php

```diff
@@ -0,0 +1,1048 @@
+    }
+
+    /**
+     * @see Plugin::manage()
+     */
+    public function manage($args, $request)
+    {
+        switch ($request->getUserVar('verb')) {
+            case 'settings':
+                $form = new CitationStyleLanguageSettingsForm($this);
+
+                if ($request->getUserVar('save')) {
+                    $form->readInputData();
+                    if ($form->validate()) {
+                        $form->execute();
+                        return new JSONMessage(true);
+                    }
+                }
+
+                $form->initData();
+                return new JSONMessage(true, $form->fetch($request));
+        }
+        return parent::manage($args, $request);
+    }
+
+    /**
+     * Route requests for the citation styles to custom page handler
+     *
+     * @see PKPPageRouter::route()
+     *
+     * @param $hookName string
+     * @param $params array
+     */
+    public function setPageHandler($hookName, $params)
+    {
+        $page =& $params[0];
+        $handler =& $params[3];
+        if ($this->getEnabled() && $page === 'citationstylelanguage') {
+            $handler = new CitationStyleLanguageHandler($this);
+            return true;
+        }
+        return false;
+    }
+
+    public function getEditorGroups(int $contextId = 0): array
+    {
+        return $this->getSetting($contextId, 'groupEditor') ?? [];
+    }
+
+    public function getTranslatorGroups(int $contextId = 0): array
+    {
+        return $this->getSetting($contextId, 'groupTranslator') ?? [];
+    }
+
+    public function getAuthorGroups(int $contextId = 0): array
+    {
+        return $this->getSetting($contextId, 'groupAuthor') ?? [];
+    }
+
+    public function getChapterAuthorGroups(int $contextId = 0): array
+    {
+        return $this->getSetting($contextId, 'groupChapterAuthor') ?? [];
+    }
+
+    public function getSerialNumber(Publication $publication): array
+    {
+        $serialNumber = [];
+        $publicationFormats = $publication->getData('publicationFormats');
+        /** @var PublicationFormat $publicationFormat */
+        foreach ($publicationFormats as $publicationFormat) {
+            if ($publicationFormat->getIsApproved()) {
+                $identificationCodes = $publicationFormat->getIdentificationCodes();
+                $identificationCodes = $identificationCodes->toArray();
+                foreach ($identificationCodes as $identificationCode) {
+                    $serialNumber[] = htmlspecialchars($identificationCode->getValue());
+                }
+            }
+        }
+        return $serialNumber;
+    }
+
+    protected function addSeriesInformation(stdClass $citationData, Publication $publication): stdClass
+    {
+        $seriesId = $publication->getData('seriesId');
+        if ($seriesId) {
+            /** @var SeriesDAO $seriesDao */
+            $seriesDao = DAORegistry::getDAO('SeriesDAO');
+            if (null !== $seriesDao) {
```

Comment:
The `$seriesDao` will always be set, so no need for the `null` check. (In OJS, trying to get the `SeriesDAO` will cause a fatal error.)